site/src/modules/dashboard/Navbar/MobileMenu.tsx

```diff
@@ -0,0 +1,339 @@
+import { cn } from "utils/cn";
+import { sortProxiesByLatency } from "./proxyUtils";
+
+const itemStyles = {
+	default: "px-9 h-10 no-underline",
+	sub: "pl-12",
+	open: "text-content-primary",
+};
+
+type MobileMenuProps = MobileMenuPermissions & {
+	proxyContextValue?: ProxyContextValue;
+	user?: TypesGen.User;
+	supportLinks?: readonly TypesGen.LinkConfig[];
+	docsHref: string;
+	onSignOut: () => void;
+	isDefaultOpen?: boolean; // Useful for storybook
+};
+
+type MobileMenuPermissions = {
+	canViewDeployment: boolean;
+	canViewOrganizations: boolean;
+	canViewAuditLog: boolean;
+	canViewHealth: boolean;
+};
+
+export const MobileMenu: FC<MobileMenuProps> = ({
+	isDefaultOpen,
+	proxyContextValue,
+	user,
+	supportLinks,
+	docsHref,
+	onSignOut,
+	...permissions
+}) => {
+	const [open, setOpen] = useState(isDefaultOpen);
+	const hasSomePermission = Object.values(permissions).some((p) => p);
+
+	return (
+		<DropdownMenu open={open} onOpenChange={setOpen}>
+			{open && (
+				<div className="fixed inset-0 top-[72px] backdrop-blur-sm z-10 bg-content-primary/50" />
+			)}
+			<DropdownMenuTrigger asChild>
+				<Button
+					aria-label={open ? "Close menu" : "Open menu"}
+					size="icon"
+					variant="ghost"
+					className="ml-auto md:hidden [&_svg]:size-6"
+				>
+					{open ? (
+						<XIcon className="size-icon-lg" />
+					) : (
+						<MenuIcon className="size-icon-lg" />
+					)}
+				</Button>
+			</DropdownMenuTrigger>
+			<DropdownMenuContent
+				className="w-screen border-0 border-b border-solid p-0 py-2"
+				sideOffset={17}
+			>
+				<ProxySettingsSub proxyContextValue={proxyContextValue} />
+
+				{hasSomePermission && (
+					<>
+						<DropdownMenuSeparator />
+						<AdminSettingsSub {...permissions} />
+					</>
+				)}
+				<DropdownMenuSeparator />
+				<DropdownMenuItem asChild className={itemStyles.default}>
+					<a href={docsHref} target="_blank" rel="noreferrer norefereer">
+						Docs
+					</a>
+				</DropdownMenuItem>
+				<DropdownMenuSeparator />
+				<UserSettingsSub
+					user={user}
+					supportLinks={supportLinks}
+					onSignOut={onSignOut}
+				/>
+			</DropdownMenuContent>
+		</DropdownMenu>
+	);
+};
+
+type ProxySettingsSubProps = {
+	proxyContextValue?: ProxyContextValue;
+};
+
+const ProxySettingsSub: FC<ProxySettingsSubProps> = ({ proxyContextValue }) => {
+	const selectedProxy = proxyContextValue?.proxy.proxy;
+	const latency = selectedProxy
+		? proxyContextValue?.proxyLatencies[selectedProxy?.id]
+		: undefined;
+	const [open, setOpen] = useState(false);
+
+	if (!selectedProxy) {
+		return null;
+	}
+
+	return (
+		<Collapsible open={open} onOpenChange={setOpen}>
+			<CollapsibleTrigger asChild>
+				<DropdownMenuItem
+					className={cn(itemStyles.default, open ? itemStyles.open : "")}
+					onClick={(e) => {
+						e.preventDefault();
+						setOpen((prev) => !prev);
+					}}
+				>
+					Workspace proxy settings:
+					<span className="leading-[0px] flex items-center gap-1">
```

Comment:
I don't think we want to set the line height to 0. It'll cause a lot of text rendering issues the moment the text has to wrap around

99% sure we want to set this to `leading-none` to set the text solid (line height of 1)

Reply:
Noted! I’m not sure why, but when I used a line height of 1, it was misaligned, and I only managed to fix it by setting it to 0. I’ll double-check that.